aws/eip.go

```diff
@@ -0,0 +1,56 @@
+package aws
+
+import (
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/gruntwork-io/cloud-nuke/logging"
+	"github.com/gruntwork-io/gruntwork-cli/errors"
+)
+
+// Returns a formatted string of EIP allocation ids
+func getAllEIPAddresses(session *session.Session, region string, excludeAfter time.Time) ([]*string, error) {
```

Comment:
Weirdly enough, this is the one resource that doesn't have a created time info attached to it

Reply:
That's... quite a problem. That means if we were to delete these, cloud-nuke could delete EIPs that are currently in use in automated tests, manual testing, etc.

I'm trying to think of workarounds...

1. We could  make `cloud-nuke` stateful. Each time it runs, it records what EIPs it saw and a timestamp of "first seen" for that EIP in DynamoDB. Next time it runs, it finds the previous entries and deletes anything it "first saw" longer ago than `excludeAfter`. This would work for any resource that doesn't track a creation date.

1. Same as above, except instead of using DynamoDB, aws-nuke adds a "first-seen" timestamp as a tag (`cloud-nuke-first-seen-timestamp`) on the EIP. This would work for any resource that doesn't have a Creation Date, but does have tags.

#2 seems fairly elegant to me...

Reply:
Option (2) is definitely the better way to go. Having to use a DB brings in another point of failure